core/http-auth-spi/src/main/java/software/amazon/awssdk/http/auth/spi/HttpSignRequest.java

```diff
@@ -56,28 +47,38 @@ static <T> Builder<T> builder(Class<T> payloadType) {
     Optional<PayloadT> payload();
 
     /**
-     * Returns the property that the {@link HttpSigner} can use during signing.
+     * Returns the identity.
+     */
+    IdentityT identity();
```

Comment:
Imo, I'd rather not parameterize this types and just do `Optional<?> payload()` and `Identity identity()` here. Subtypes would override them with the more-specific return type.

This makes it so that code that deal with the base sign request can just deal with a `SignRequest` instead of a `SignRequest<?, ?>`, and the base builder can just deal with a `Builder<?> instead of a Builder<?, ?, ?>`.

WDYT?

Reply:
I think code that deals with base sign request, might benefit from the type, atleast the IdentityT. So the BearerTokenSigner can extract the token from `HttpSignRequest<?, TokenIdentity identity>`. Though I can see removing the PayloadT (will reduce it by one), but means the subtypes have to (re)define the `payload()` method instead of specifying the generic in type definition. And as an interface, having a HttpSignRequest that has a `payload()` accessor but doesn't know it's type doesn't feel great to me, so I'm in favor of keeping them.

Reply:
Cool. And we can reassess at the surface area review as well, once we can see the impact on the signers, etc.